crates/dbsp/src/operator/group/topk.rs

```diff
@@ -32,6 +35,28 @@ where
     }
 }
 
+impl<K, V, R> Stream<RootCircuit, OrdIndexedZSet<K, V, R>>
+where
+    K: DBData,
+    V: DBData,
+    R: DBWeight + ZRingValue,
+{
+    /// Pick `k` smallest values in each group based on a custom comparison
+    /// function.
+    ///
+    /// This method is similar to [`topk_asc`](`Stream::topk_asc`), but instead
+    /// of ordering elements according to `trait Ord for V`, it uses a
+    /// user-defined comparison function `F`.
+    pub fn topk_custom_order<F>(&self, k: usize) -> Self
+    where
+        F: CmpFunc<V>,
+    {
+        self.map_index(|(k, v)| (k.clone(), <WithCustomOrd<V, F>>::new(v.clone())))
```

Comment:
So this is how I could have implemented it.
I was thinking that you would push the packaging into the group transformed to save 2 stages.
A composition of group transformers is a group transformer itself.